packages/@n8n/db/src/migrations/common/1763048000000-ActivateExecuteWorkflowTriggerWorkflows.ts

```diff
@@ -0,0 +1,166 @@
+ * or an errorTrigger node. Also disables any other trigger nodes within those workflows.
+ */
+export class ActivateExecuteWorkflowTriggerWorkflows1763048000000 implements ReversibleMigration {
+	async up({ escape, runQuery, runInBatches, parseJson }: MigrationContext) {
+		const tableName = escape.tableName('workflow_entity');
+		const idColumn = escape.columnName('id');
+		const versionIdColumn = escape.columnName('versionId');
+		const nodesColumn = escape.columnName('nodes');
+		const activeColumn = escape.columnName('active');
+		const activeVersionIdColumn = escape.columnName('activeVersionId');
+
+		const inactiveWorkflows = `SELECT ${idColumn}, ${nodesColumn}, ${versionIdColumn}, ${activeVersionIdColumn} FROM ${tableName} WHERE ${activeColumn} = false`;
+
+		await runInBatches<Workflow>(inactiveWorkflows, async (workflows) => {
+			for (const workflow of workflows) {
+				const nodes = parseJson(workflow.nodes);
+
+				const executeWorkflowTriggerNode = nodes.find(
+					(node: Node) => node.type === EXECUTE_WORKFLOW_TRIGGER_NODE_TYPE,
+				);
+
+				const errorTriggerNode = nodes.find((node: Node) => node.type === ERROR_TRIGGER_NODE_TYPE);
+
+				if (!executeWorkflowTriggerNode && !errorTriggerNode) {
+					continue;
+				}
+
+				if (executeWorkflowTriggerNode) {
+					const inputSource = executeWorkflowTriggerNode.parameters?.inputSource;
+
+					const shouldActivateByInputSource =
+						inputSource === 'passthrough' || inputSource === 'jsonExample';
+
+					// For legacy nodes (no inputSource), check if they have parameters defined
+					let hasLegacyParameters = false;
+					if (!inputSource) {
+						const workflowInputs = executeWorkflowTriggerNode.parameters?.workflowInputs;
+						hasLegacyParameters = Boolean(
+							workflowInputs &&
+								typeof workflowInputs === 'object' &&
+								'values' in workflowInputs &&
+								Array.isArray(workflowInputs.values) &&
+								workflowInputs.values.length > 0 &&
+								this.hasValidWorkflowInputs(workflowInputs.values),
+						);
+					}
+
+					if (!shouldActivateByInputSource && !hasLegacyParameters) {
+						continue;
+					}
+				}
+
+				// Disable other trigger nodes
+				let nodesModified = false;
+				nodes.forEach((node: Node) => {
+					// Check if node is a trigger (excluding executeWorkflowTrigger, and errorTrigger only when no valid executeWorkflowTrigger exists)
+					if (
+						node.type !== EXECUTE_WORKFLOW_TRIGGER_NODE_TYPE &&
+						(node.type !== ERROR_TRIGGER_NODE_TYPE || executeWorkflowTriggerNode) &&
+						this.isTriggerNode(node.type)
+					) {
+						if (!node.disabled) {
+							node.disabled = true;
+							nodesModified = true;
+						}
+					}
+				});
+
+				if (nodesModified) {
+					await runQuery(
+						`UPDATE ${tableName} SET ${activeColumn} = :active, ${nodesColumn} = :nodes, ${activeVersionIdColumn} = :versionId WHERE ${idColumn} = :id`,
+						{
+							active: true,
+							nodes: JSON.stringify(nodes),
```

Comment:
This now makes the nodes in `workflow_entity` and `workflow_history` out of sync. As long as we have the data duplicated in both, we need to keep them in sync or the data is inconsistent